src/main/java/service/impl/ShopServiceImpl.java

```diff
@@ -0,0 +1,47 @@
+package service.impl;
+
+import database.Storage;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import model.FruitTransaction;
+import service.ShopService;
+import strategy.OperationStrategy;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    public OperationStrategy getOperationStrategy() {
```

Comment:
You have getter for private field operationStrategy, but it's never used. It's a good practice to remove unused code. This helps to make your code cleaner and easier to read.